File: frontend/src/types/plan.ts
export interface Plan {
  id: number
  date: string
  type: 'meal' | 'exercise'
  name: string
  calories: number
  duration?: number
  actual_duration?: number
  completed: boolean
  created_at?: string
  updated_at?: string
}

export interface DailyPlans {
  date: string
  meals: Plan[]
  exercises: Plan[]
  total_items: number
}

export interface DailyStats {
  date: string
  total_calories_intake: number
  total_calories_burned: number
  net_calories: number
  total_exercise_duration: number
  total_items: number
  completed_items: number
  completion_rate: number
}

File: backend/app/models/plan.py
"""
Plan model for storing diet and exercise plans.
"""
from sqlalchemy import Column, Integer, String, Float, Boolean, Date, DateTime
from datetime import datetime
from app.database import Base


class Plan(Base):
    """Model for diet and exercise plan items."""
    
    __tablename__ = "plans"
    
    id = Column(Integer, primary_key=True, index=True)
    date = Column(Date, nullable=False, index=True)
    type = Column(String(10), nullable=False, index=True)  # 'meal' or 'exercise'
    name = Column(String(200), nullable=False)
    calories = Column(Float, nullable=False)
    duration = Column(Integer)  # minutes, for exercise only
    completed = Column(Boolean, default=False)
    actual_duration = Column(Integer)  # actual time spent
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
